spec/controllers/application_controller_spec.rb

```diff
@@ -12,89 +12,103 @@
+          expect(controller.most_focus('category', nil)).to be_empty
         end
+      end
 
-        it "returns a hash of size 2" do
-          new_category1 = create(:category, userid: user1.id)
-          new_category2 = create(:category, userid: user1.id)
-          new_moment = create(:moment, userid: user1.id, category: Array.new(1, new_category1.id))
-          new_strategy = create(:strategy, userid: user1.id, category: Array.new(1, new_category2.id))
-          result = controller.most_focus('category', user1.id)
-          expect(result.length).to eq(2)
-          expect(result[new_category1.id]).to eq(1)
-          expect(result[new_category2.id]).to eq(1)
-        end
+      context 'when categories exist' do
+        subject { controller.most_focus('category', user_id) }
 
-        it "returns a correct hash of size 3" do
-          new_category1 = create(:category, userid: user1.id)
-          new_category2 = create(:category, userid: user1.id)
-          new_category3 = create(:category, userid: user1.id)
-          new_category4 = create(:category, userid: user1.id)
-          new_moment = create(:moment, userid: user1.id, category: Array.new(1, new_category2.id))
-          new_strategy = create(:strategy, userid: user1.id, category: [new_category1.id, new_category2.id, new_category3.id, new_category4.id])
-          result = controller.most_focus('category', user1.id)
-          expect(result.length).to eq(3)
-          expect(result[new_category1.id]).to eq(1)
-          expect(result[new_category2.id]).to eq(2)
-          expect(result[new_category3.id]).to eq(1)
-          expect(result[new_category4.id]).to eq(nil)
-        end
+        context 'when the same category is used twice' do
+          before do
+            create(:moment, category: [category.id], userid: user_id)
+            create(:strategy, category: [category.id], userid: user_id)
+          end
 
-        it "returns a correct hash of size 1 belonging to another user" do
-          new_category1 = create(:category, userid: user2.id)
-          new_category2 = create(:category, userid: user2.id)
-          new_category3 = create(:category, userid: user2.id)
-          new_category4 = create(:category, userid: user2.id)
-          new_moment = create(:moment, userid: user2.id, category: Array.new(1, new_category2.id), viewers: Array.new(1, user1.id), published_at: Time.zone.now)
-          new_strategy = create(:strategy, userid: user2.id, category: [new_category1.id, new_category2.id, new_category3.id, new_category4.id], published_at: Time.zone.now)
-          result = controller.most_focus('category', user2.id)
-          expect(result.length).to eq(1)
-          expect(result[new_category1.id]).to eq(nil)
-          expect(result[new_category2.id]).to eq(1)
-          expect(result[new_category3.id]).to eq(nil)
-          expect(result[new_category4.id]).to eq(nil)
+          it 'includes duplicate categories once' do
```

Comment:
This test is unclear to me, could you clarify this in the test? It's not clear how the two expects imply that the duplicate categories appear once.

Reply:
The test is meant to demonstrate that if I create a `Moment` and a `Strategy` and give each one the same category, that the category will only appear once in the hash returned by `#most_focus`. I _think_ that was part of the intent in the original test.

However, now that you mention it, I think the first `expect` statement is quite unclear. I should assert that `subject.length == 1`. Following on from that, the actual value of `subject[category.id]` should be 2 since the one category has appeared twice.

Reply:
Thanks for clarifying and let's do what you suggested!